lib/fastlane/plugin/wpmreleasetoolkit/actions/android/android_generate_apk_from_aab.rb

```diff
@@ -0,0 +1,132 @@
+        rescue StandardError
+          UI.user_error!('bundletool is not installed. Please install it using the instructions at https://developer.android.com/studio/command-line/bundletool.')
+          raise
+        end
+
+        # If no AAB param was provided, attempt to get it from the lane context
+        # First GRADLE_ALL_AAB_OUTPUT_PATHS if only one
+        # Second GRADLE_AAB_OUTPUT_PATH if it is set
+        # Else use the specified parameter value
+        if params[:aab_file_path].nil?
+          all_aab_paths = Actions.lane_context[SharedValues::GRADLE_ALL_AAB_OUTPUT_PATHS] || []
+          aab_file_path = if all_aab_paths.count == 1
+                            all_aab_paths.first
+                          else
+                            Actions.lane_context[SharedValues::GRADLE_AAB_OUTPUT_PATH]
+                          end
+        else
+          aab_file_path = params[:aab_file_path]
+        end
+
+        # If no AAB file path was found, raise an error
+        if aab_file_path.nil?
+          UI.user_error!('No AAB file path was specified and none was found in the lane context. Please specify the `aab_file_path` parameter or ensure that the relevant build action has been run prior to this action.')
+          raise
+        end
+
+        apk_output_file_path = params[:apk_output_file_path]
+        keystore_path = params[:keystore_path]
+        keystore_password = params[:keystore_password]
+        keystore_key_alias = params[:keystore_key_alias]
+        signing_key_password = params[:signing_key_password]
+
+        sh(generate_command(aab_file_path, apk_output_file_path, keystore_path, keystore_password, keystore_key_alias, signing_key_password))
+      end
+
+      #####################################################
+      # @!group Documentation
+      #####################################################
+
+      def self.description
+        'Generates an APK from the specified AAB'
+      end
+
+      def self.details
+        'Generates an APK from the specified AAB'
+      end
+
+      def self.available_options
+        [
+          FastlaneCore::ConfigItem.new(
+            key: :aab_file_path,
+            env_name: 'ANDROID_AAB_FILE_PATH',
+            description: 'The path to the AAB file. If not speicified, the action will attempt to read from the lane context using the `SharedValues::GRADLE_ALL_AAB_OUTPUT_PATHS` and `SharedValues::GRADLE_AAB_OUTPUT_PATH` keys',
+            type: String,
+            optional: true,
+            default_value: nil,
+            verify_block: proc { |p| UI.user_error!("AAB path `#{p}` is not a valid file path.") unless File.file?(p) }
+          ),
+          FastlaneCore::ConfigItem.new(
+            key: :apk_output_file_path,
+            env_name: 'ANDROID_APK_OUTPUT_PATH',
+            description: 'The output path where the APK file will be generated. The directory will be created if it does not yet exist',
```

Comment:
This needs to be updated.

The behavior that you implemented so far (the one which adds `mv` command at the end of `bundletool` invocation), expects this to be the path to the APK (and not a folder), and does not create the directory if it doesn't exist.

Even my rewriting of your behavior (using a `Dir.mktmpdir` + `arguments` array and `sh('bundletool', *arguments)`), which uses the exact same assumptions as your code expects the path to be to the APK and not to a folder, and doesn't create the intermediate folders if they don't exist. So even if you go with my suggestion, that description isn't true anymore.

And to be honest I prefer this parameter to behave like your or my latest implementation is doing, i.e. only support that path to be a path to the `.apk` (and not to a folder), and also expecting that the parent folder of that destination already exists. This is better practice to not create the intermediate folders if they don't exist imho, so that a typo in the path doesn't end up creating the file in an unexpected place.

```suggestion
            description: 'The path to the output APK file we want to generate',
```